add Downgrade CI

kylebeggs opened the pull request:
I *think* I did this correctly with the help of Claude explaining to me, but I'm still not 100% on how this package works. Please suggest edits!

closes #509

kylebeggs committed:
add Downgrade CI

oxinabox commented on src/plugins/ci.jl:
```diff
@@ -417,7 +470,7 @@ function collect_versions(t::Template, versions::Vector)
 end
 
-const AllCI = Union{AppVeyor, GitHubActions, TravisCI, CirrusCI, GitLabCI, DroneCI}
+const AllCI = Union{AppVeyor, GitHubActions, TravisCI, CirrusCI, GitLabCI, DroneCI, DowngradeCI}
```
this isn't really a CI system like the others.
I do not think it be belongs here or in this file.

It's more its own thing. like the thing for building documentation.

I suggest also renaming it: to like "DowngradeDepencencyTests"

kylebeggs committed:
make downgrade like documenter, etc.

kylebeggs committed:
Merge branch 'master' into master

kylebeggs committed:
Merge branch 'master' into master

oxinabox commented on src/plugins/downgrade.jl:
```diff
@@ -0,0 +1,51 @@
+        "HAS_SKIP" => !isempty(p.skip),
+    )
+    git = getplugin(t, Git)
+    if git !== nothing
```
should we error without this?

kylebeggs replied:
I don't think so. Other plugins in this package don't error for similar situations.

kylebeggs replied:
Good catch. I switched to the same pattern used by `TravisCI`, `GitLabCI`, `Codecov`, and `Coveralls`:

```julia
"BRANCH" => something(default_branch(t), DEFAULT_DEFAULT_BRANCH),
```

This always populates `BRANCH` (falling back to `"main"` when the `Git` plugin is absent) instead of silently omitting it. Erroring would have been inconsistent with every other CI/coverage plugin in the package, so this felt like the closer fit. Pushed in ed2b8f8.

kylebeggs committed:
Merge branch 'master' into master

kylebeggs committed:
fix(downgrade): always populate BRANCH via default_branch fallback

Match the pattern used by TravisCI, GitLabCI, Codecov, and Coveralls:
fall back to DEFAULT_DEFAULT_BRANCH when the Git plugin is absent
instead of silently omitting BRANCH from the view dict.

kylebeggs committed:
test(downgrade): regenerate reference fixtures for DowngradeDependencyTests

Adds the Downgrade.yml fixtures for the AllPlugins and WackyOptions
reference test cases, and updates the WackyOptions README.md fixture
to include the Downgrade badge.

kylebeggs committed:
docs(downgrade): include DowngradeDependencyTests in user manual

Documenter aborts with missing_docs because the plugin's docstring
isn't referenced from any @docs block, blocking the Documentation
CI job.

kylebeggs committed:
test(downgrade): cover needs_username for DowngradeDependencyTests
